### DISCUSSION TITLE
Mega menu

### DISCUSSION THREAD

**Discussion by finalwebsites** [bootscore/bootscore#832] [Ideas]
Hi,

I got the question regarding a mega menu from a customer.
There are plugins which build a MM for WordPress, but for simple cases it might a good idea to implement this kind of stuff into the theme.
Any idea how to implement a BS walker menu "callback" for a HTML structure like here?
https://bootstrap-menu.com/detail-megamenu.html

**Comment by crftwrk** [comment]
Bootstrap doesn‘t support a Mega Menu and the nav walker used here https://github.com/AlexWebLab/bootstrap-5-wordpress-navbar-walker doesn‘t supports this for this reason as well.

However, nav walker is a pluggable function which can be simply overridden in child https://github.com/bootscore/bootscore/blob/main/inc/navwalker.php. Maybe there is a way to edit the walker to get a Mega Menu.

My customers sometimes are asking for a Mega Menu as well. I solved this with custom classes in the menu items and some jQuery. Not so nice but works, check https://intelligent-heizen.info/ (Desktop view).

**Reply by finalwebsites** [reply]
The mega menu looks not bad, but I need to add also some additional content. Thanks anyway, I will check your work and show you the result (when I have some). I'm also working on my own site, which is using BS5 but not your theme (started the theme years ago and updated the files 1000 times)

**Reply by crftwrk** [reply]
Here is an interesting approach for a Mega Menu walker using widgets in dropdowns. Didn't tested it and it's for Bootstrap 4, but this should be quickly adapted to Bootstrap 5. https://github.com/jaycbrf4/WP-Bootstrap4-MegaMenu-Navwalker
